### PR TITLE
Deprecate react-tools

### PR DESCRIPTION
**Pull request by zpao**
It's time to just use jstransform.

Still a bit to do here (including testing it). I think it would be good to actually redo some of the build process around packaging things for npm.

cf. #3743

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -8,88 +8,29 @@
...
-  return options;
+  // Copy out the values we need and make sure they are compatible with
+  // jstransform options.
+  var options = {
+    harmony: opts.harmony,
+    stripTypes: opts.stripTypes,
+    sourceMapInline: opts.sourceMap,
+    sourceFilename: opts.sourceFilename,
+    es6module: opts.es6module,
+    nonStrictEs6module: opts.nonStrictEs6module
```
Comment (spicyj) [inline main.js]: target?

**Comment by spicyj** [comment]
Looks good, I think.

File: bin/jsx-internal [hunk 2]
```diff
@@ -2,7 +2,7 @@
 // -*- mode: js -*-
 "use strict";
 
-var transform = require('../main').transform;
+var transform = require('jstransform/simple').transform
```
Comment (chicoxyzzy) [inline bin/jsx-internal]: semicolon is missing here

File: main.js (hunk 1 above)
Comment (zpao) [inline main.js]: Good call.

**Comment by zpao** [comment]
All right, all updated here. I tested that JSXTransformer still works (and made a small change while I was there to get rid of a dependency on jstransform/src/inline-source-map). Biggest change in the last updates here was to ensure we always set the react options when calling into jstransform, to maintain parity with expectations that we had before.

cc @Daniel15 @rmosolgo @kmeht since this might impact your uses of JSXTransformer outside of the browser. We should be effectively equivalent but wanted you to know ahead of time.

**Comment by Daniel15** [comment]
Thanks for the heads up, I'll do some testing with ReactJS.NET.

**Comment by zpao** [comment]
If you don't want to build yourself, you can download from
http://react.zpao.com/builds/master/latest/.

**Pull request merged**